static/js/task/controllers/task.controller.js

```diff
@@ -22,17 +22,22 @@
             self.task_worker_id = $routeParams.taskWorkerId;
             self.task_id = $routeParams.taskId;
 
-            if (!self.task_worker_id) { //if they navigate away mid-queue and then attempt task from task feed
```

Comment:
I checked this and it looks fine to me. Will need more elaborate comment on how it is breaking as you might be expecting something else to happen.

My understanding is this - 
a) User can select tasks and click on Finish Tasks, which clears the queue and adds selected tasks to queue
b) User can  work on queue as collection of tasks in order
c) In case user jumps to a different view when queue is not empty (without refresh), queue is still in place but user has no way to go back. So user will have to reselect tasks to continue. (This section can be improved where we allow worker to see the queue and restart from where he left).
d) There is a limit on number of tasks worker can save and keep in progress state.